static/js/narrow.js

```diff
@@ -296,10 +302,34 @@ export function activate(raw_operators, opts) {
             // the stream/topic pair that was requested to some other
             // location, then we should retarget this narrow operation
             // to where the message is located now.
-            const adjusted_operators = adjusted_operators_if_moved(raw_operators, target_message);
-            if (operators_changed) {
-                activate(adjusted_operators, opts);
-                return;
+            const narrow_topic = filter.operands("topic")[0];
+            const narrow_stream_name = filter.operands("stream")[0];
+            const narrow_stream_id = stream_data.get_sub(narrow_stream_name).stream_id;
+            const narrow_dict = {stream_id: narrow_stream_id, topic: narrow_topic};
+
+            const narrow_matches_target_message = util.same_stream_and_topic(
+                target_message,
+                narrow_dict,
+            );
```

Comment:
Why do we need the `narrow_matches_target_message` check?  I would have thought that `adjusted_operators` would be `null` in this case ... or is this important to avoid unnecessary work in cases where the message was moved somewhere and then moved back?

Reply:
I'm adding a comment with this explanation, since it seems correct to me.

Reply:
Basically, my thought was that if the message matches the narrow, then we know no redirect is necessary. And we have to get the narrow stream/topic values to do the edit history loop, so it seemed logical to check the current message values that as well as the edit history.

But yes, it would return `null` in that situation.

I think you could go back to the structure that @amanagr had in that `adjusted_parameters_if_moved` is always run for a target message and the logic check of the edit history and page parameters happens inside the function.

Reply:
I think this implementation is fine as written -- we just should have a comment explaining what we're doing so nobody is tempted to delete it as unnecessary when reading this code later :).